packages/react-next/src/components/SearchBox/SearchBox.base.tsx

```diff
@@ -26,27 +28,29 @@ const useComponentRef = (
 };
 
 export const SearchBoxBase = React.forwardRef((props: ISearchBoxProps, forwardedRef: React.Ref<HTMLDivElement>) => {
-  const [hasFocus, { setTrue: setHasFocusTrue, setFalse: setHasFocusFalse }] = useBoolean(false);
+  const [hasFocus, setHasFocus] = React.useState(false);
   const [value = '', setValue] = useControllableValue(props.value, props.defaultValue, props.onChange);
   const rootElementRef = React.useRef<HTMLDivElement>(null);
   const inputElementRef = React.useRef<HTMLInputElement>(null);
   const mergedRootRef = useMergedRefs(rootElementRef, forwardedRef);
-  const fallbackId = useId(COMPONENT_NAME);
+  const id = useId(COMPONENT_NAME, props.id);
 
   const {
     ariaLabel,
-    placeholder,
     className,
     disabled,
     underlined,
     styles,
     // eslint-disable-next-line deprecation/deprecation
     labelText,
+    // eslint-disable-next-line deprecation/deprecation
+    placeholder = labelText,
     theme,
-    clearButtonProps = { ariaLabel: 'Clear text' },
+    clearButtonProps = defaultClearButtonProps,
     disableAnimation = false,
```

Comment:
let's move these to `SearchBoxBase.defaultProps` instead

Reply:
We don't really have a standard pattern around default props for non-`composed` function components (or even our class components) right now, so I think it's better to leave this as-is.

Reply:
i just learned defaultProps is getting deprecated https://github.com/yannickcr/eslint-plugin-react/issues/2396